### PR TITLE
Add support for using pkgdepends for provisioner

### PR DESCRIPTION
**Pull request by richfitz**

### CODE REVIEW

File: R/configure.R
Comment (weshinsley) [inline R/configure.R]: Is it better to say args$policy <- args$policy %||% "lazy" ? 
(I didn't know x <- x <- ...  did that)

options I think are "lazy" or "upgrade" in pkgdepends - do we want to fail here if it's not one of those, or leave that to pkgdepends to complain about? (Prob the latter?)

File: tests/testthat/test-pkgdown.R
Comment (weshinsley) [inline tests/testthat/test-pkgdown.R]: Should this file be called test-pkgdepends.R ?

File: tests/testthat/test-pkgdown.R
Comment (weshinsley) [inline tests/testthat/test-pkgdown.R]: What should the behaviour be if we have multiple repos? 
(Currently, we end up with repos=c("repo1", "repo2") and refs = c("ref1", "ref2")

Are we able to express that one ref comes from one repo, and another from another? (Eg, in the case where we have different versions of packages in different repos, and we might pick and mix - which is all very terrible, but malariasimulation and friends seem to have that issue.)

Comment (weshinsley) [reply tests/testthat/test-pkgdown.R]: One other thing - I guess a particular branch of a package is ref1@branch1 - and is treated in exactly the same way (hence no need for a song+dance about it)

Comment (weshinsley) [reply tests/testthat/test-pkgdown.R]: And (finally!) - is there any value in checking for duplicate refs (with/without `@branch`) ?

**Review by weshinsley — CHANGES_REQUESTED** [review]
OK - I think I understand this pretty much - a couple of comments on the way through - and maybe one test file to rename before merging?

File: R/configure.R
Comment (richfitz) [reply R/configure.R]: fixed now. The latter we're not touching and will direct users to pkgdepends in the (eventual) docs

File: tests/testthat/test-pkgdown.R
Comment (richfitz) [reply tests/testthat/test-pkgdown.R]: yes

File: tests/testthat/test-pkgdown.R
Comment (richfitz) [reply tests/testthat/test-pkgdown.R]: as discussed, this is all bits that pkgdepends will try and do the right thing on, we'll try not to get too involved with its logic

**Review by weshinsley — APPROVED** [review]
Great

**Pull request merged**
